specification/purview/data-plane/Azure.Analytics.Purview.SelfServicePolicies/preview/2022-12-01-preview/purviewSelfServicePolicy.json

```diff
@@ -0,0 +1,348 @@
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "An error response received from the Policy Service",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string",
+                "description": "The error code"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SelfServicePolicy_Delete": {
+            "$ref": "./examples/SelfServicePolicy_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SelfServicePolicyDecisionRule": {
+      "type": "object",
+      "description": "The decision rule of a SelfServicePolicy",
+      "required": [
+        "effect",
+        "purviewRoleName",
+        "resourceAzurePath"
+      ],
+      "properties": {
+        "effect": {
+          "description": "The effect for rule",
+          "enum": [
+            "Permit"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "Decision",
+            "modelAsString": true
+          }
+        },
+        "purviewRoleName": {
+          "type": "string",
+          "description": "This is field will indicate the role of a self service policy."
+        },
+        "principal": {
+          "type": "string",
+          "description": "Azure Active Directory Id for users"
+        },
+        "principalGroup": {
+          "type": "string",
+          "description": "Azure Active Directory Id for groups"
+        },
+        "resourceAzurePath": {
+          "type": "string",
+          "description": "The resource path of the azure resource"
+        }
+      }
+    },
+    "ErrorModel": {
+      "description": "The error model for policy",
+      "required": [
+        "code",
+        "message"
+      ],
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code",
+          "minLength": 1,
+          "type": "string"
+        },
+        "message": {
+          "description": "The error message",
+          "minLength": 1,
+          "type": "string"
+        },
+        "details": {
+          "description": "The error details",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorModel"
+          }
+        }
+      }
+    },
+    "ErrorResponseModel": {
+      "description": "The error response model for policy",
+      "required": [
+        "error"
+      ],
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ErrorModel"
+        }
+      }
+    },
+    "SelfServicePolicy": {
+      "description": "The Self-service policy model",
+      "required": [
+        "decisionRule",
+        "workflowRunId"
+      ],
+      "type": "object",
+      "properties": {
+        "workflowRunId": {
```

Comment:
You should not call this property `workflowRunId` or describe it as the id of a workflow. This is the id of the selfServicePolicy so it should be called something like `policyId`. Your service might derive it from the workflowRunId but this is an internal detail that should not leak out into the API contract.

Reply:
Have modified the field to policyId instead of workflowRunId.